src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -153,6 +153,28 @@
 phone_extension_type = CLIArgumentType(
     help='The extension of the customer service number of the publisher. Only digits are allowed and number of digits should not exceed 10.')
 
+kind_type = CLIArgumentType(
+    help="The kind of topic resource.",
+    arg_type=get_enum_type(['azure', 'azurearc']),
```

Comment:
Not sure if you remember `arg_type` will not work within `CLIArgumentType`...

Reply:
i think it was a problem specific to state flag? this works. 

---
In reply to: [603736909](https://github.com/Azure/azure-cli/pull/17496#discussion_r603736909) [](ancestors = 603736909)